bgpd/bgp_addpath.c

```diff
@@ -185,26 +207,24 @@ int bgp_addpath_tx_path(enum bgp_addpath_strat strat,
 static void bgp_addpath_flush_type(struct bgp *bgp, afi_t afi, safi_t safi,
 				   enum bgp_addpath_strat addpath_type)
 {
-	struct bgp_node *rn;
-	struct bgp_path_info *pi;
+	struct bgp_node *rn, *nrn;
 
 	for (rn = bgp_table_top(bgp->rib[afi][safi]); rn;
 	     rn = bgp_route_next(rn)) {
-		idalloc_drain_pool(
-			bgp->tx_addpath.id_allocators[afi][safi][addpath_type],
-			&(rn->tx_addpath.free_ids[addpath_type]));
-		for (pi = bgp_node_get_bgp_path_info(rn); pi; pi = pi->next) {
-			if (pi->tx_addpath.addpath_tx_id[addpath_type]
-			    != IDALLOC_INVALID) {
-				idalloc_free(
-					bgp->tx_addpath
-						.id_allocators[afi][safi]
-							      [addpath_type],
-					pi->tx_addpath
-						.addpath_tx_id[addpath_type]);
-				pi->tx_addpath.addpath_tx_id[addpath_type] =
-					IDALLOC_INVALID;
-			}
+		if (safi == SAFI_MPLS_VPN) {
```

Comment:
Don't we need more than MPLS_VPN safi?  What about the other table types that are dual level?

Reply:
The two other SAFIs that need special treatment are ENCAP and EVPN, but ENCAP is deprecated (not sure why it wasn't removed yet since the encap commands no longer exist) and EVPN doesn't support add-path at this moment (not sure if it makes sense for it).